ivy/functional/ivy/activations.py

```diff
@@ -77,13 +79,35 @@ def gelu(
     return current_backend(x).gelu(x, approximate=approximate, out=out)
 
 
+@handle_device_shifting
+def _leaky_relu_jax_like(
+    x: Union[ivy.Array, ivy.NativeArray],
+    /,
+    *,
+    fn_original: Optional[Callable],
+    alpha: float = 0.2,
+    out: Optional[ivy.Array] = None,
+) -> ivy.Array:
+    return ivy.where(
+        (
+            ivy.logical_or(
+                ivy.real(x) < 0, ivy.logical_and(ivy.real(x) == 0, ivy.imag(x) < 0)
+            )
+        ),
+        x * alpha,
+        x,
+    )
+
+
 @handle_exceptions
 @handle_nestable
 @handle_array_like_without_promotion
 @handle_out_argument
 @to_native_arrays_and_back
 @handle_array_function
+@handle_complex_input
 @handle_device_shifting
+@add_attributes(jax_like=_leaky_relu_jax_like)
```

Comment:
Do we really need this decorator? 🤔 

Cant we just do `leaky_relu.jax_like = _leaky_relu_jax_like` below this function.

Reply:
I tried that first, it took me a while to come up with the decorator as a workaround. `functools.wraps` copies attributes *up* the stack of decorators, but not down, so if you do the assignment after all of the decorators then only the top-level one is able to see it. The assignment has to run after the function definition but before the complex number handling decorator, hence the new decorator here

Reply:
I don't fully follow, I tried doing `leaky_relu.jax_like = _leaky_relu_jax_like` after the function implementation and removing the `add_attributes` decorator completely. It works fine.

Or perhaps you're saying there could be some other issues with this approach? 🤔